modules/nf-core/sourmash/taxannotate/main.nf

```diff
@@ -0,0 +1,51 @@
+process SOURMASH_TAXANNOTATE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::sourmash=4.5.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.5.0--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.5.0--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(gather_results)
+    path(taxonomy)
```

Comment:
Most likely, nf-core modules for sourmash will be used together (in a chain).
Currently, the output of the `gather` module is a single file.
Therefore, it's easier to provide it as input to `tax annotate`  via `--gather-csv`.
It would be nice to support `--from-file` as well. But it will require manipulating (`collect`ing) a channel that contains multiple gather results in the workflow. So this solution is not very general and depends on a project specifics.

So I think it's easier to use just a single input now.

Reply:
Technically we should support all input files as per the guidelines, however indeed from the descriptions `--from-file` does not sound particularly 'nextflow-y' so I'm going to avert my eyes from this for now ;)